src/esp/gfx/GenericDrawable.cpp

```diff
@@ -183,6 +187,39 @@ void GenericDrawable::draw(const Mn::Matrix4& transformationMatrix,
     shader_->bindObjectIdTexture(*(materialData_->objectIdTexture));
   }
 
+  if (skinData_) {
+    // Gather joint transformations
+    auto& skin = skinData_->skinData->skin;
+    auto& jointIdToArticulatedObjectNodes =
+        skinData_->jointIdToArticulatedObjectNode;
+    auto& transformNodes = skinData_->jointIdToTransformNode;
+    Cr::Containers::Array<Mn::Matrix4> jointTransformations{
+        Cr::NoInit, skin->joints().size()};
+
+    // Undo root node transform so that the model origin matches the root
+    // articulated object link.
+    auto invRootTransform =
+        jointIdToArticulatedObjectNodes[skinData_->rootJointId]
+            ->absoluteTransformationMatrix()
+            .inverted();
+
+    Mn::Matrix4 lastTransform = Mn::Matrix4{Magnum::Math::IdentityInit};
+    for (std::size_t i = 0; i != jointTransformations.size(); ++i) {
+      auto jointNodeIt = transformNodes.find(skin->joints()[i]);
+      if (jointNodeIt != transformNodes.end()) {
+        jointTransformations[i] =
+            invRootTransform *
+            jointNodeIt->second->absoluteTransformationMatrix() *
+            skin->inverseBindMatrices()[i];
+        lastTransform = jointTransformations[i];
+      } else {
+        // Joint not found - use last transform.
+        jointTransformations[i] = lastTransform;
```

Comment:
Why does this failed-find happen in practice?

Reply:
This happens if the URDF doesn't perfectly match the skinned model.
For example, it's somewhat common for skinned models to have extra leaf bones.

Our rig-to-URDF tool scripts should prevent this from happening.

Reply:
In that case, can we do a loud error on startup instead of this silent failure?